src/main/java/org/example/DAO/imp/ProductDao.java

```diff
@@ -0,0 +1,139 @@
+package org.example.DAO.imp;
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.TransactionRequiredException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.CriteriaUpdate;
+import javax.persistence.criteria.Root;
+import lombok.extern.slf4j.Slf4j;
+import org.example.DAO.Dao;
+import org.example.model.Product;
+
+/** Class responsible for database operations at product entity. */
+@Slf4j
+public class ProductDao implements Dao<Product> {
+
+  EntityManagerFactory etf;
```

Comment:
No need for this to be a member/attribute of the class.

Reply:
Done.